### PR TITLE
[TTreeReader][ROOT-10148] Allow to read arrays of Double32_t and Float16_t with TTreeReader

### PR DESCRIPTION
**Pull request by dpiparo**

### CODE REVIEW

File: tree/treeplayer/src/TTreeReaderArray.cxx [hunk 1]
```diff
@@ -460,7 +461,7 @@ void ROOT::Internal::TTreeReaderArrayBase::CreateProxy()
          fProxy = 0;
          return;
       }
-      if (!branchActualType) {
+      if (!branchActualType && !isOpaqueTypedef) {
```
Comment (pcanal) [inline tree/treeplayer/src/TTreeReaderArray.cxx]: Isn't that too coarse?  What about is the type is opaque and the user still pass a different type (eg. a long or a float for Double32_t)?

Comment (dpiparo) [reply tree/treeplayer/src/TTreeReaderArray.cxx]: In this case we are left with string comparison I am afraid.

Comment (dpiparo) [reply tree/treeplayer/src/TTreeReaderArray.cxx]: not really. See below

**Comment by dpiparo** [comment]
after some changes, this should be fine. I also added a new tutorial and simplified `double32.C`

**Comment by dpiparo** [comment]
@phsft-bot build please to see what happens with Win

**Comment by Axel-Naumann** [comment]
@phsft-bot build just on windows10/default

**Pull request merged**